sched/jobDefSerializer_test.go

```diff
@@ -0,0 +1,249 @@
+package sched
+
+import (
+	"testing"
+	//"github.com/leanovate/gopter"
+	//"github.com/leanovate/gopter/prop"
+	"fmt"
+	"time"
+	"github.com/leanovate/gopter"
+	"github.com/leanovate/gopter/prop"
+)
+
+const serializeType = BinarySerialize
+
+
+func Test_JobDefCompare(t *testing.T) {
+	origJobDef := makeSampleJobDefintion()
+
+	// validate comparison logic
+	if ok, msg := origJobDef.Equal(origJobDef); !ok {
+		panic(fmt.Sprintf("error: comparison logic doesn't work comparing an object against itself %s\n", msg))
+	}
+	var actualJobDef JobDefinition
+	if ok, msg := origJobDef.Equal(actualJobDef); ok {
+		panic(fmt.Sprintf("error: comparison logic doesn't work comparing an object against nil %s\n", msg))
+	}
+
+}
+
+func Test_NilJobDefinition(t *testing.T) {
+	if ok, err := Serialize(nil, serializeType); err == nil {
+		panic("error: should have gotten an error from serializing a nil object\n")
+		fmt.Printf(string(ok))
+	}
+}
+
+func Test_FixedJobDefinition(t *testing.T) {
+	// use this test to test a specific jobDefinition struct
+	origJobDef := makeSampleJobDefintion()
+
+
+	// serialize the object to a byte array
+	if asByteArray, err := Serialize(&origJobDef, serializeType); err != nil {
+		panic(fmt.Sprintf("error: couldn't serialize the fixed job def. %s\n", err.Error()))
+	} else {
+		// deserialize the byte array
+		if newJobDef, err := Deserialize(asByteArray, serializeType); err != nil {
+			panic(fmt.Sprintf("error: deserializing the byte Array: %s\n", string(asByteArray)) + err.Error())
+			// compare the orig and generated job types
+		} else if (origJobDef.JobType != newJobDef.JobType) {
+			panic(fmt.Sprintf("fail: deserialized job def type doesn't equal original job def, expected %s, got %s\n", origJobDef.JobType, newJobDef.JobType))
+			// compare the orig and generated task definitions
+		} else if ok, msg := origJobDef.Equal(*newJobDef); !ok {
+			panic(fmt.Sprintf("fail: task definitions are not equal: %s\n", msg))
+		} else {
+			fmt.Printf(fmt.Sprintf("ok=%t\n", ok))
+		}
+	}
+}
+
+func makeSampleJobDefintion() JobDefinition {
+	jobDef := JobDefinition{}
+	jobDef.Tasks = make(map[string]TaskDefinition)
+	jobDef.JobType = "jobTypeVal"
+	taskDefinition := TaskDefinition{}
+	taskDefinition.SnapshotId = "snapshotIdVal"
+	tx, _ := time.ParseDuration("+3s")
+	taskDefinition.Timeout = tx
+	envVars := make(map[string]string)
+	taskDefinition.EnvVars = envVars
+	envVars["var1"] = "var1Value"
+	envVars["var2"] = "var2Value"
+	args := []string{"arg1", "arg2"}
+	taskDefinition.Argv = args
+	jobDef.Tasks["task0"] = taskDefinition
+	//Print(jobDef)  -I enable this for debugging
+
+	return jobDef
+}
+
+
+func (this *JobDefinition) Equal(actualJobDef JobDefinition) (bool, string) {
```

Comment:
Files named _test.go are included only in the compile for tests. It's like a build directive. So putting it in _test.go (even if in the same package, which, yes, my bad, I've been doing it the wrong way) will include it for tests but not in the binary.

It's fine if we want it in general; I'm -0 on it just because it's creeping featurism and it's not super clear why you'd need it, and definitions of equality often end up confusing later down the road.